### ISSUE TITLE
Would be good to have DeepUndefinable

### ISSUE THREAD

**Issue by ShravanSunder** [sindresorhus/type-fest#612]
See https://github.com/ts-essentials/ts-essentials/tree/master/lib/deep-undefinable

is it ok to put a PR to add something like this?

**Comment by sindresorhus** [comment]
Yes, makes sense to have such a type. It should be named `UndefinableDeep` though.
